### ISSUE TITLE
Noisy logging of prometheus when not used.

### ISSUE THREAD

**Issue by sjmudd** [vitessio/vitess#4149]
#### Overview of the Issue

I see this in my vtctld log file. (and I think in others too)

```
I0822 16:39:20.646249   16378 prometheusbackend.go:71] PrometheusUnsupportedMetricType: Not exporting to Prometheus an unsupported metric type of *stats.String: BuildUser
I0822 16:40:20.646517   16378 throttled.go:77] PrometheusUnsupportedMetricType: skipped 7 log messages
```

vttablet shows something similar:

```
I0822 11:38:36.213150   10612 prometheusbackend.go:71] PrometheusUnsupportedMetricType: Not exporting to Prometheus an unsupported metric type of *stats.String: BuildGitRev
I0822 11:39:36.213418   10612 throttled.go:77] PrometheusUnsupportedMetricType: skipped 19 log messages
```

I am not using prometheus so the logging here seems noisy and unnecessary.

#### Reproduction Steps

* start vtctld and look at log file.
* start vttablet and look at the log file.

#### Binary version

I'm running current code from 08/2018.

#### Operating system and Environment details

Running on Linux x86_64, bare metal (not relevant I think)

#### Log Fragments

See above.

#### Suggestion to fix

If the prometheus usage is not configured then don't run and don't generate log messages.

**Comment by demmer** [comment]
Oops. I actually fixed this ages ago in #3901 but never merged. This should solve the issue.

**Comment by sjmudd** [comment]
thanks.
